Modules/_winapi.c

```diff
@@ -1672,6 +1677,10 @@ _winapi_GetShortPathName_impl(PyObject *module, LPCWSTR path)
         PyErr_SetFromWindowsErr(0);
     }
     return result;
+#else
+    PyErr_SetString(PyExc_OSError, "GetShortPathName unavailable on this platform");
```

Comment:
Your call here, but if you put the ifdef around the entire function definition and regenerate the clinic headers (should be `python Tools\clinic\clinic.py Modules\_winapi.c` unless someone finally moved the clinic.py script) then it should entirely omit the function when you build.

Or maybe it'd make sense just to return the original string again? I'm not sure where this function gets used, but in general, unconditional errors are harder to detect than a missing function, and are less convenient than sensible-but-technically-incorrect behaviours. This is an internal module, so we only really need to consider stdlib uses, though of course we want to avoid breaking existing third-party code as well.

Reply:
As far as I can tell we don't use this anywhere